frontend/src/components/AuthUtils/index.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Utility functions to get various parameters corresponding to the currently
+ * authenticated user. These functions should ONLY be called from private pages,
+ * i.e. the user must be authenticated to ensure the parameters can be
+ * retrieved.
+ */
+
+import { AuthUtilsConsumer, getCurrentUser } from './AuthUtilsConsumer.js';
+import { SIGN_IN } from '../../constants/routes.js';
+
+/**
+ * Called when one of the authentication utility functions are used when the
+ * user has not logged in yet. Since the utility functions require that they be
+ * called from private pages, this function being called implies that the user
+ * managed to access a private page without being authenticated first. So they
+ * are immiedately redirected to the SIGN_IN page.
+ */
+function redirectToSignIn() {
+  window.location.href = SIGN_IN;
+}
+
+/**
+ * @returns {String} The user's display name.
+ */
+export function getUserDisplayName() {
+  if (!getCurrentUser()) {
+    redirectToSignIn();
+    return null;
+  }
```

Comment:
These lines are repeated in several functions below. It would be good to separate them out into a standard function that can be called by all the functions below.

You could then call it like:
```
if (!userLoggedIn()) { return null; }
return ...
```

Reply:
Ah yeah that would probably be cleaner for code reuse, I abstracted that piece of code into its own function.